### ISSUE TITLE
Imported interfaces destructured in function argument gives no-shadow

### ISSUE THREAD

**Issue by umidbekkarimov** [eslint/typescript-eslint-parser#545]
Error occurred after upgrade from `typescript-eslint-parser^20.0.0`.
Potentially related to #540.

**What version of TypeScript are you using?**

3.1.6

**What version of `typescript-eslint-parser` are you using?**

20.1.1

**What code were you trying to parse?**

```ts
import { Foo } from "./Foo";

function foo(args: Foo) {} // No error
function bar([...args]: Foo) {} // No error
function baz({ ...args }: Foo) {} //  'Foo' is already declared in the upper scope

const foo = Foo; // No error
const [...foo] = Foo; // No error
const { ...foo } = Foo; // No error
```

**Comment by mysticatea** [comment]
Thank you for the report.

This looks the same problem as #535.
It should be fixed by #540.
